src/Application/Controller/Project/ReportController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Controller\Project;
+
+use App\DB\Entity\Project\Program;
+use App\DB\Entity\Project\ProgramInappropriateReport;
+use App\DB\Entity\User\User;
+use App\Project\Event\ReportInsertEvent;
+use App\Project\ProjectManager;
+use App\User\UserManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ReportController extends AbstractController
+{
+    public function __construct(
+        private readonly UserManager $user_manager,
+        private readonly ProjectManager $project_manager,
+        private readonly TranslatorInterface $translator,
+        private readonly EventDispatcherInterface $event_dispatcher,
+        private readonly AuthorizationCheckerInterface $authorization_checker,
+        private readonly TokenStorageInterface $usage_tracking_token_storage,
+        private readonly EntityManagerInterface $entity_manager,
+    ) {
+    }
+
+    #[Route(path: '/api/reportProject/reportProject.json', name: 'api_report_program', defaults: ['_format' => 'json'], methods: ['POST', 'GET'])]
```

Comment:
This is still the legacy api - use the new system for it: https://github.com/Catrobat/Catroweb-Api
Once you have defined the specification, i can help you generate the boiler code and include it in this project, to add your logic